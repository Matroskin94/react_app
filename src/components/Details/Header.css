.title {
    display: inline-block;
}

.favoriteButton {
    display: inline-block;
    position: absolute;
    margin: 50px 20px;
    right: 0px;
}

@media(--mediumScreen) {
    .favoriteButton {
        margin-right: 50px;
    }
}