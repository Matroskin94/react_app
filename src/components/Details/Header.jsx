import React from 'react';

const Header = props => (
    <div>
        <h1>Details Header</h1>
    </div>
);

export default Header;
