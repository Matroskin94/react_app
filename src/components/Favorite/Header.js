import React from 'react';

const Header = props => ({
    render: function() {
        return (
            <div>
                <h3 className = 'textCenter'>Favourites</h3>
            </div>
        );
    }
});

export { Header };
