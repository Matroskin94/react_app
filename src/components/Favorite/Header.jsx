import React from 'react';

const Header = props => (
    <div>
        <h3 className='textCenter'>Favourites</h3>
    </div>
);

export default Header;
