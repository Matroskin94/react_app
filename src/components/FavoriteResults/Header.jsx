import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
    <div data-test='favHeader'>
        <Link to='/' data-test='favesBackLink'> BACK </Link>
        <h3 className='textCenter'>Favourites</h3>
    </div>
);

export default Header;
