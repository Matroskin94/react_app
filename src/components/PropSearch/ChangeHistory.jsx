import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import queryString from 'query-string';

export default url => ConnectHistory => {
    class ChangeHistory extends PureComponent {
        static contextTypes = {
            router: PropTypes.object.isRequired
        };
        static propTypes = {
            location: PropTypes.object
        }
        static defaultProps = {
            location: {}
        }

        getURLParams = () => queryString.parse(this.props.location.search)

        historyPush = coordinates => {
            this.context.router.history.push(`${url}${coordinates}`);
        }

        render() {
            return <ConnectHistory
                {...this.props}
                historyPush={this.historyPush}
                getURLParams={this.getURLParams}
            />;
        }
    }

    return ChangeHistory;
};
