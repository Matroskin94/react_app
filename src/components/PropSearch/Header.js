import React from 'react';

const Header = props => ({
    render: function() {
        return (
            <div>
                <button className='button' id='faves-but'>Faves</button>
                <h3>Property cross</h3>
            </div>
        );
    }
});

export { Header };
