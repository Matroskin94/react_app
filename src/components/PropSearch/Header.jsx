import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
    <div>
        <Link to='/results/favorite' >Faves </Link><br />
        <h3>Property cross</h3>
    </div>
);

export default Header;
