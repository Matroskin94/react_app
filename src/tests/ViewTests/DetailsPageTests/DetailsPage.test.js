import React from 'react';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

import Header from '../../../components/Details/Header.jsx';
import ChangeHistory from '../../../components/PropSearch/ChangeHistory.jsx';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
	context: {
		router: {
			history: {
				push: jest.fn()
			}
		}
	},
	router: {
		history: {
				push: jest.fn()
		}
	},
	subscribe: jest.fn(),
	dispatch: jest.fn(),
	clearActions: jest.fn(),
	replaceReducer: jest.fn(),
	getState: jest.fn(),
});

describe('Details page testing', () => {
	it('Details header test', () => {
		const onFavoriteClickMock = jest.fn();
		const historyBackMock = jest.fn();
		// Необходимо прокинуть контекст с ним работает компонент обёртка
		const wrapper = shallow(
				<Provider store={store}>
					<Header
						handleFavoriteClick={onFavoriteClickMock}
						isFavorite="true"
						historyBack={historyBackMock}
					/>
				</Provider>
			);

		expect(wrapper.prop('isFavorite')).toEqual('true');
		expect(wrapper.prop('handleFavoriteClick')).toEqual(onFavoriteClickMock);
		expect(wrapper).toMatchSnapshot();
	});
});