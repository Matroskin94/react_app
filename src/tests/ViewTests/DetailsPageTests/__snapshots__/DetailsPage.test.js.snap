// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page testing Details header test 1`] = `
<Header
  handleFavoriteClick={[MockFunction]}
  historyBack={[MockFunction]}
  isFavorite={true}
/>
`;
